xbmc/cores/VideoPlayer/VideoRenderers/OverlayRendererGUI.cpp

```diff
@@ -207,6 +216,14 @@ void COverlayText::Render(OVERLAY::SRenderState &state)
   // clamp inside screen
   y = std::max(y, (float) res.Overscan.top);
   y = std::min(y, res.Overscan.bottom - m_height);
+  
+  // draw background box if not transparent
+  UTILS::Color bgcolor = m_layout->GetBgColour();
+  if (bgcolor != UTILS::COLOR::TRANSPARENT)
+  {
+    CRect backgroundbox(x - (m_layout->GetTextWidth())* 0.55f, y, x + m_layout->GetTextWidth()* 0.55f, y + m_layout->GetTextHeight());
+    CGUITexture::DrawQuad(backgroundbox, bgcolor);
```

Comment:
that can't be done on VP rendering pipeline. CGUITexture::DrawQuad is slow, it does sync CPU / GPU

Reply:
Thanks for looking. Can you provide any alternative suggestion?

Reply:
look into shaders, i.e. gl_shader_frag_fonts.glsl
with some simple modification it can draw a background without any performance penalty.

Reply:
disregard, DrawQuad is actually called with nullptr for texture